### DISCUSSION TITLE
Pass the selection limit of 999

### DISCUSSION THREAD

**Discussion by mirisbowring** [photoprism/photoprism#2786] [Q&A] [answered]
Hi,

i just imported an older folder with around 5k Photos which are belonging to an holiday week.
Now, i switched the sort order in timeline view to "added recently" and selected all images in order to put them into an album.

Unfortunately, photoprism is telling me that i have passed the 999 selection limit which makes it very difficult to ensure that every photo is added (i have to continue select/add where i stopped before).

Any ideas on how to proceed with such case? How are you doing large imports?

**Comment by lastzero** [comment, marked as answer]
You can generally select complete folders and add them to albums without having to select individual pictures. Since you imported the files, they have been renamed / moved. That means in your special case, there is no folder you could clone / select - although the original path is remembered in case the files were in a sub folder when you imported them. Because adding thousands of pictures to a manually managed album doesn't scale well, the solution we are going to implement are albums based on search filters (basically like Moments). All that's missing is a UI. Check GitHub Issues for the related feature request and discussions as this topic came up a few times before.

**Comment by lastzero** [comment]
Although you can select more than 1000 pictures in iCloud, it seems that you can't do anything with them 🤣

![Screenshot 2023-03-28 at 12 16 10](https://user-images.githubusercontent.com/301686/228206484-038193c5-1ba8-491b-a96f-e2aa6276402f.png)
